applications/StructuralMechanicsApplication/custom_io/gid_eigen_io.h

```diff
@@ -0,0 +1,281 @@
+    template< class TVarType>
+    inline void WriteEigenResults( ModelPart& rModelPart, 
+                                   const TVarType& rVariable,
+                                   std::string Label, 
+                                   const SizeType AnimationStepNumber );
+      
+      
+    ///@}
+    ///@name Access
+    ///@{ 
+    
+    
+    ///@}
+    ///@name Inquiry
+    ///@{
+    
+    
+    ///@}      
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const
+    {
+        std::stringstream buffer;
+        buffer << "GidEigenIO" ;
+        return buffer.str();
+    }
+    
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const {rOStream << "GidEigenIO";}
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const {}
+    
+        
+    ///@}      
+    ///@name Friends
+    ///@{
+    
+        
+    ///@}
+      
+  protected:
+    ///@name Protected static Member Variables 
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Protected member Variables 
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Protected Operators
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Protected Operations
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Protected  Access 
+    ///@{ 
+    
+    
+    ///@}      
+    ///@name Protected Inquiry 
+    ///@{ 
+    
+    
+    ///@}    
+    ///@name Protected LifeCycle 
+    ///@{ 
+    
+        
+    ///@}
+    
+  private:
+    ///@name Static Member Variables 
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Member Variables 
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Private Operators
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Private Operations
+    ///@{ 
+    
+    
+    ///@} 
+    ///@name Private  Access 
+    ///@{ 
+    
+    
+    ///@}    
+    ///@name Private Inquiry 
+    ///@{ 
+    
+    
+    ///@}    
+    ///@name Un accessible methods 
+    ///@{ 
+
+    
+    ///@}    
+    
+}; // Class GidEigenIO 
+
+  ///@} 
+  
+  ///@name Type Definitions       
+  ///@{ 
+
+template<>
+inline void GidEigenIO::WriteEigenResults<Variable<double>>(ModelPart& rModelPart, 
+                                                  const Variable<double>& rVariable,
+                                                  std::string Label, 
+                                                  const SizeType AnimationStepNumber) 
+{
+    Label += "_" + rVariable.Name();  
+    GiD_fBeginResult( mResultFile, (char*)Label.c_str() , "EigenVector_Animation",
+                        AnimationStepNumber, GiD_Scalar,
+                        GiD_OnNodes, NULL, NULL, 0, NULL );
+
+    for (const auto& node : rModelPart.Nodes())
+    {
+        const double& nodal_result = node.FastGetSolutionStepValue(rVariable);
+        GiD_fWriteScalar( mResultFile, node.Id(), nodal_result );
+    }
+
+    GiD_fEndResult(mResultFile);
+}
+
+template<>
```

Comment:
Could the template be removed and just overload the function since there are two separate functions here?

Reply:
done